code/controllers/UserController.php

```diff
@@ -17,8 +17,9 @@ public function index()
     {
         if (!Member::currentUserID()) {
             return $this->redirect('/Security/login');
-        }
-        return $this->renderWith(array('UserController', 'Page'));
+        } else if (UserInvitation::create()->canCreate(Member::CurrentUser()) == false){
```

Comment:
I see. Which isn't easy to override from RequestHandler. 
Okay, so because the index action is the only one we need secure, I think it's fine to then:
* remove the check from index in `$allowed_actions`
* scrap the permissionCheck method I mentioned above and replace the existing signed in member check with 
```php
if(!Permission::check('ACCESS_USER_INVITATIONS')){
   return Security::permissionFailure();
}
```
which does both.

Agree?